crates/sui-framework/packages/dwallet-system/sources/eth_dwallet.move

```diff
@@ -51,12 +51,11 @@ module dwallet_system::eth_dwallet {
         eth_state: &EthereumState,
     ): vector<MessageApproval> {
         // Validate that the Ethereum dWallet capability is for the correct network.
-        let latest_state_network = ethereum_state::get_latest_ethereum_state_network(latest_ethereum_state);
-        assert!(eth_dwallet_cap.network == latest_state_network, ENetworkMismatch);
+        assert!(eth_dwallet_cap.latest_ethereum_state_id == object::id(latest_ethereum_state), EStateObjectMismatch);
 
         // Validate that the EthereumState is for the correct network.
-        let eth_state_network = ethereum_state::get_ethereum_state_network(eth_state);
-        assert!(eth_state_network == latest_state_network, ENetworkMismatch);
+        let eth_state_id = object::id(eth_state);
+        assert!(eth_state_id == ethereum_state::get_ethereum_state_id_from_latest(latest_ethereum_state), EStateObjectMismatch);
```

Comment:
this checks that the latest state is used, I think we said we don't have to. Lets do the opposite check: that the `latest_ethereum_state_id == eth_state.latest_ethereum_state_id`